controllers/topics/index.js

```diff
@@ -84,13 +85,32 @@ const putFollowTopic = async (req, res) => {
 
 const getTopics = async (req, res) => {
   const {name} = req.query;
+  const signUserId = req.userId;
+  const anonymousId = await getAnonymUser(req.userId);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

These lines are retrieving the signed-in user's ID and the anonymous user's ID. However, it seems like you're passing the signed-in user's ID (`req.userId`) to the `getAnonymUser` function. If this function is supposed to get an anonymous user's ID, it might not need any parameters or it should take a different parameter that identifies the session or request, not a signed-in user's ID.
```diff
- const anonymousId = await getAnonymUser(req.userId);
+ const anonymousId = await getAnonymUser(req.sessionId); // assuming req.sessionId exists and can be used to identify an anonymous user
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->